azurerm/internal/services/netapp/resource_arm_netapp_volume.go

```diff
@@ -316,15 +365,28 @@ func expandArmNetAppVolumeExportPolicyRule(input []interface{}) *netapp.VolumePr
 			allowedClients := strings.Join(*utils.ExpandStringSlice(v["allowed_clients"].(*schema.Set).List()), ",")
 			cifsEnabled := v["cifs_enabled"].(bool)
 			nfsv3Enabled := v["nfsv3_enabled"].(bool)
-			nfsv4Enabled := v["nfsv4_enabled"].(bool)
+			nfsv41Enabled := v["nfsv41_enabled"].(bool)
+
+			// TODO: Remove in 2.0 and replace with nfsv41Enabled := v["nfsv41_enabled"].(bool)
+			// var nfsv41Enabled bool
+
+			// _, nfsv4Ok := v["nfsv4_enabled"].(bool)
+
+			// if nfsv4Ok {
+			// 	nfsv41Enabled = v["nfsv4_enabled"].(bool)
+			// } else {
+			// 	// TODO: Keep only this line in 2.0
+			// 	nfsv41Enabled = v["nfsv41_enabled"].(bool)
+			// }
```

Comment:
This may be tricker then i thought due to it being a set, it might not be possible with a bunch of bools as there is no way to determine which one is set or both are set. Going to a list of protocols would work, otherwise might have to make this change in 2.0

Reply:
Yes, this sounds a good idea, we can create the object with all items set as false and whatever protocol is in the list we make the change in the object to true.